src/types.ts

```diff
@@ -1,8 +1,11 @@
 export type CftRequest = {
+    console: ConsoleObject;
     spyLibrary?: SupportedSpyLibrary;
     testFramework?: SupportedTestFramework;
 };
 
 export type SupportedSpyLibrary = "fallback" | "jasmine" | "jest" | "sinon" | unknown;
 
 export type SupportedTestFramework = "mocha" | "jasmine" | "jest" | unknown;
+
+export type ConsoleObject = { [P in keyof Console]?: Console[P] extends Function ? boolean : never };
```

Comment:
Hmm, I'm starting to second-guess `ConsoleObject` as a name, since "Object" doesn't really add much. How about something like `ConsoleSettings` or `ConsoleMethods`?

Reply:
Changed